### PR TITLE
fix: :bug: dry runs shouldn't be creating labels

### PR DESCRIPTION
**Pull request by zolamk**
## Description

- Made sure labels aren't being created for dry runs

## Related issue

Closes Issue #33

## Type of change

Bug fix (non-breaking change which fixes an issue)

## How was this tested?

I created a pull request on my repo and then using the cli
1. I first did a normal run and checked If the labels were created on the pull request ✅
2. removed the created labels ✅
3. then I did a dry run and checked no labels were created on the pull request ✅

## Checklist

- [x] I have performed a self-review of my code
- [x] I have ran `task check -f` and have no issues

### CODE REVIEW

**Comment by reviewpad-team** [comment]
**Reviewpad Report**

**:information_source: Messages**
* @marcelosousa: you are being notified because critical code was modified

File: engine/exec.go [hunk 1]
```diff
@@ -67,20 +67,26 @@ func Eval(file *ReviewpadFile, env *Env) (*Program, error) {
 			labelName = label.Name
 		}
 
-		labelExists, err := checkLabelExists(env, labelName)
-		if err != nil {
-			return nil, err
-		}
+		// don't check for existence
+		// or create label for dry runs
+		if !env.DryRun {
```
Comment (marcelosousa) [inline engine/exec.go:73]: Please remove this empty line.

File: engine/env.go [hunk 2]
```diff
@@ -45,6 +46,7 @@ func NewEvalEnv(
 	collector collector.Collector,
 	pullRequest *github.PullRequest,
 	interpreter Interpreter,
+	dryRun bool,
```
Comment (marcelosousa) [inline engine/env.go:49]: Since the dryRun is a 'context' variable, I suggest that we make it the second argument.

File: engine/env.go [hunk 3]
```diff
@@ -36,6 +36,7 @@ type Env struct {
 	Collector   collector.Collector
 	PullRequest *github.PullRequest
 	Interpreter Interpreter
+	DryRun      bool
```
Comment (marcelosousa) [inline engine/env.go:39]: Similar comment as below.

**Review by marcelosousa — COMMENTED** [review]
Thank you for this contribution 🙌 It's looking quite good. I've made a few small comments.

**Comment by marcelosousa** [comment]
@zolamk please check my last comment in the diff - thank you!

**Review by marcelosousa — APPROVED** [review]
LGTM 👍

**Comment by marcelosousa** [comment]
@zolamk can you please take a look at the conflicts? Thanks!

**Comment by marcelosousa** [comment]
@zolamk please take a look at the issues reported at https://github.com/reviewpad/reviewpad/pull/152#issuecomment-1189511534

**Comment by zolamk** [comment]
Closing this in favor of PR #154 because I messed up the git history.

**Pull request closed without merging**